### PR TITLE
docs: describe switch from kafka connect api to kafkaconnectors

### LINKED ISSUE #2884: [Doc] Consequences and migration path for enabling KafkaConnectors CRDs is missing
**Suggestion / Problem**
After enabling `KafkaConnectors` CRD through the annotation as described on the documentation 
we lost all the configuration from the existing connectors configured through the REST API

Our expectation was that both could live together

**Documentation Link**
- https://strimzi.io/docs/master/#kafkaconnector_resources
- https://strimzi.io/docs/master/#con-creating-managing-connectors-str
- https://strimzi.io/docs/master/#proc-enabling-kafkaconnectors-deployment-configuration-kafka-connect

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

Adds a section to the _deploying_ guide that describes how to switch from using the Kafka Connect API to using KafkaConnectors. 

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

This PR fixes #2884

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
Nice, thanks.

**Comment by katheris** [comment]
I've had a read through the docs and generally it looks great. There were two small changes I wondered about. Firstly, is it worth saying explicitly that the annotation "strimzi.io/use-connector-resources" needs to be set to "true"? I know the other docs mention that but it would only be a small change to mention it here. Secondly we could link to the Kafka Connect REST API docs, incase the reader isn't familiar with it. The link is https://kafka.apache.org/documentation.html#connect_rest

File: documentation/modules/deploying/con-deploy-kafka-connect-config-connector-api.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+.Switching from the Kafka Connect API to using KafkaConnectors
+
+You can switch from using the Kafka Connect API to using KafkaConnectors to manage your connectors.
+To make the switch, do the following in the the order shown:
+
+. Deploy `KafkaConnector` resources with the configuration to create your connector instances.
+. Enable KafkaConnectors in your Kafka Connect configuration using the `strimzi.io/use-connector-resources` annotation.
++
+If you enable KafkaConnectors before creating the resources, you will delete all your connectors.
```
Comment (tombentley) [inline documentation/modules/deploying/con-deploy-kafka-connect-config-connector-api.adoc:57]: Should this be a `WARNING:` (or whatever the syntax is)?

**Review by tombentley — APPROVED** [review]

File: documentation/modules/deploying/con-deploy-kafka-connect-config-connector-api.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/modules/deploying/con-deploy-kafka-connect-config-connector-api.adoc:57]: Changed to a _WARNING_

**Comment by PaulRMellor** [comment]
> I've had a read through the docs and generally it looks great. There were two small changes I wondered about. Firstly, is it worth saying explicitly that the annotation "strimzi.io/use-connector-resources" needs to be set to "true"? I know the other docs mention that but it would only be a small change to mention it here. Secondly we could link to the Kafka Connect REST API docs, incase the reader isn't familiar with it. The link is https://kafka.apache.org/documentation.html#connect_rest

Thanks @katheris. I've updated as suggested.

**Pull request merged**